packages/components/src/scroll-to-top-button/index.tsx

```diff
@@ -0,0 +1,70 @@
+import styled from '@emotion/styled';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect, useState } from 'react';
+import Gridicon from '../gridicon';
+
+type ScrollButtonProps = {
+	classname?: string;
+	children?: React.ReactNode;
+	scrollThreshold?: number;
+};
+
+const ScrollButton = styled( Button )< { visible: boolean } >`
+	position: fixed;
+	display: flex;
+	opacity: ${ ( props ) => ( props.visible ? 1 : 0 ) };
+	align-items: center;
+	justify-content: center;
+	inset-block-end: 24px;
+	inset-inline-start: 24px;
+	z-index: 176;
+	height: 42px;
+	width: 42px;
+	background-color: #000;
+	color: #fff;
+	border-radius: 4px;
+	transition: opacity 0.3s ease-in-out;
+`;
+
+export const ScrollToTopButton = ( {
+	classname,
+	children,
+	scrollThreshold = window.innerHeight,
+}: ScrollButtonProps ) => {
+	const [ visible, setVisible ] = useState( false );
+
+	const scrollCallback = useCallback( () => {
+		if ( window.scrollY > scrollThreshold ) {
+			setVisible( true );
+		} else {
+			setVisible( false );
+		}
+	}, [ scrollThreshold ] );
+
+	useEffect( () => {
+		window.addEventListener( 'scroll', scrollCallback );
+
+		return () => {
+			window.removeEventListener( 'scroll', scrollCallback );
+		};
+	}, [ scrollCallback, classname, children ] );
+
+	const scrollToTop = () => {
+		window.scrollTo( {
+			top: 0,
+			behavior: 'smooth',
+		} );
+	};
```

Comment:
How about creating a hook instead of a visual component? This way we could abstract the functionality and use it in both `client/sites-dashboard/components/sites-dashboard.tsx` and `packages/help-center/src/components/back-to-top-button.tsx`. I think something like this would work great:

```typescript
interface UseScrollToTopOptions< T > {
	scrollTargetRef: React.MutableRefObject< T | null >;
	smoothScrolling: boolean;
	isBelowThreshold: ( scrollTarget: T ) => boolean;
}

const useScrollToTop = < T extends HTMLElement >( {
	scrollTargetRef,
	smoothScrolling,
	isBelowThreshold,
}: UseScrollToTopOptions< T > ) => {
	const [ isButtonVisible, setVisible ] = useState( false );

	const scrollCallback = useCallback( () => {
		if ( scrollTargetRef.current ) {
			setVisible( isBelowThreshold( scrollTargetRef.current ) );
		}
	}, [ scrollTargetRef, isBelowThreshold ] );

	useEffect( () => {
		if ( scrollTargetRef.current ) {
			const scrollElement = scrollTargetRef.current;

			scrollElement.addEventListener( 'scroll', scrollCallback );

			return () => {
				scrollElement.removeEventListener( 'scroll', scrollCallback );
			};
		}
	}, [ scrollTargetRef, scrollCallback ] );

	const scrollToTop = useCallback( () => {
		if ( ! scrollTargetRef.current ) {
			return;
		}

		scrollTargetRef.current.scrollTo( {
			top: 0,
			behavior: smoothScrolling ? 'smooth' : 'auto',
		} );
	}, [ scrollTargetRef, smoothScrolling ] );

	return {
		scrollToTop,
		isButtonVisible,
	};
};
```

The idea is that it should be possible to use this hook in both places as:

```typescript
const isBelowThreshold = useCallback( ( containerNode: HTMLElement ) => {
	const SCROLL_THRESHOLD = 400;

	return containerNode.scrollTop > SCROLL_THRESHOLD;
}, [] );

const { isButtonVisible, scrollToTop } = useScrollToTop( {
	scrollTargetRef: elementRef,
	isBelowThreshold,
	smoothScrolling: false,
} );
```

Making it possible to specify custom thresholds (`isBelowThreshold`) to display the button, specify smooth or instant scrolling (`smoothScrolling`), and set a custom target for the scrolling predicate through `scrollTargetRef`: it's `window` in the SMD, but it's the parent of the button element in the help center.

We should be able to proceed with this version, letting the visual elements untouched and totally decoupled with the logic 🤠 

We can pair if you're having trouble using the hook in both places 👍